ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -63,31 +67,44 @@ governor_BOOTSTRAP_PEERS_TIMEOUT = 15 * 60
 -- state) then, until the node reaches a clean state, this monitoring action
 -- will be disabled and thus churning will be disabled as well.
 --
+-- On the other hand, if Genesis mode is on for the node, this action responds
+-- to changes in ledger state judgement monitoring actions to change the static
+-- set of target peers.
 targetPeers :: (MonadSTM m, Ord peeraddr)
             => PeerSelectionActions peeraddr peerconn m
             -> PeerSelectionState peeraddr peerconn
             -> Guarded (STM m) (TimedDecision m peeraddr peerconn)
-targetPeers PeerSelectionActions{readPeerSelectionTargets}
+targetPeers PeerSelectionActions{ readPeerSelectionTargets,
+                                  readLedgerStateJudgement,
+                                  peerTargets = ConsensusModePeerTargets { praosTargets, genesisSyncTargets },
+                                  churnMutex }
             st@PeerSelectionState{
               publicRootPeers,
               localRootPeers,
               targets,
-              ledgerStateJudgement,
               bootstrapPeersFlag,
-              hasOnlyBootstrapPeers
+              hasOnlyBootstrapPeers,
+              consensusMode
             } =
     Guarded Nothing $ do
-      targets' <- readPeerSelectionTargets
-      check ( isNodeAbleToMakeProgress bootstrapPeersFlag
-                                       ledgerStateJudgement
-                                       hasOnlyBootstrapPeers
-            && targets' /= targets
-            && sanePeerSelectionTargets targets'
-            )
-      -- We simply ignore target updates that are not "sane".
-
+      targets'  <- readPeerSelectionTargets
+      ledgerStateJudgement' <- readLedgerStateJudgement
+      targets'' <- tryReadTMVar churnMutex <&>
+        \case
+          Nothing -> targets'
+          _just   -> case (consensusMode, ledgerStateJudgement') of
```

Comment:
Might as well pattern match on `Just ()`

Reply:
If we can't take the mutex we will return the default `PeerSelectionTargets`? why don't we return the current `targets` instead of `targets'` ?

Reply:
The current ```targets``` are probably out of date when the Nothing branch is matched, and we have to get the new targets to apply that churn has given us in the TVar.

Reply:
I've squashed the no-mutex commit so this is now moot anyway.